src/main/scala/org/renci/cam/QueryService.scala

```diff
@@ -397,4 +382,19 @@ object QueryService extends LazyLogging {
     SPARQLQueryExecutor.runSelectQueryAs[SlotStuff](queryText.toQuery)
   }
 
+  private def getTRAPIQEdgePredicates(edge: TRAPIQueryEdge): RIO[ZConfig[AppConfig] with HttpClient, List[IRI]] =
+    for {
+      edgeType <- ZIO.fromOption(edge.`type`).orElseFail(new Exception("failed to get edge type"))
+      queryText = sparql"""
+                   SELECT DISTINCT ?predicate
+                   WHERE {
+                     ?predicate $RDFSSubPropertyOf+ ${edgeType.iri} .
+                     FILTER NOT EXISTS { ?predicate a $BiolinkMLSlotDefinition }
+                   }
+                   """
+      querySolutions <- SPARQLQueryExecutor.runSelectQuery(queryText.toQuery)
+      predicates = querySolutions
+        .flatMap(qs => qs.varNames.asScala.filter(a => qs.contains(a)).map(v => IRI(qs.get(v).toString)))
```

Comment:
I see how this avoids runtime errors but it seems really indirect. It would be clearer to ask for the value of the `predicate` var. Or make a local case class to take advantage of sparql-utils:

```scala
case class Predicate(predicate: IRI)
for {
...
predicates <- SPARQLQueryExecutor.runSelectQueryAs[Predicate](queryText.toQuery)
...
```

Reply:
I am returning a List[IRI] and there is an implicit in the companion object to "embedInSPARQL".  You are saying that you want yet another wrapper to double up the use of sparql-utils?  Returning List[Predicate[IRI]] seems redundant.

Reply:
Sorry, I wasn't very clear. `Predicate` is just for describing the query projection. Do something like `predicates.map(_.predicate)` to return.